### DISCUSSION TITLE
PyTest module not working

### DISCUSSION THREAD

**Discussion by jahidhasanlinix** [facebookresearch/hydra#2307] [Q&A]
I did installed pytest via `pip3 install pytest` but its still showing below error when i run `python my_app.py`:
```
from pytest import raises
ModuleNotFoundError: No module named 'pytest'
```

```
$ pip3 show pytest
Name: pytest
Version: 7.0.1
Summary: pytest: simple powerful testing with Python
Home-page: https://docs.pytest.org/en/latest/
Author: Holger Krekel, Bruno Oliveira, Ronny Pfannschmidt, Floris Bruynooghe, Brianna Laugher, Florian Bruhin and others
Author-email: None
License: MIT
Location: ~/.local/lib/python3.6/site-packages
Requires: py, pluggy, iniconfig, attrs, packaging, importlib-metadata, tomli
```

I was running this code from this repo: [Code](https://github.com/facebookresearch/hydra/blob/1.2_branch/examples/tutorials/basic/your_first_hydra_app/3_using_config/my_app.py)

**Comment by Jasha10** [comment]
I suspect that `pytest` was not installed in the same virtual environment as the environment supplying your `python` executable.
To help with debugging, please paste the output from each of the following commands:
```text
which pip3
which python
which pytest
```

**Reply by jahidhasanlinix** [reply]
> I suspect that `pytest` was not installed in the same virtual environment as the environment supplying your `python` executable. To help with debugging, please paste the output from each of the following commands:
> 
> ```
> which pip3
> which python
> which pytest
> ```
Here is the output of those commands:
```
$ which pip3
/usr/bin/pip3
$ which python
/usr/bin/python
$ which pytest
/home/jh/.local/bin/pytest
```

**Reply by jahidhasanlinix** [reply]
Able to fixed it. Create a new conda env for this and it works now.
